rtv/content.py

```diff
@@ -367,52 +385,96 @@ def __init__(self, name, submissions, loader, order=None):
             raise exceptions.SubredditError('No submissions')
 
     @classmethod
-    def from_name(cls, reddit, name, loader, order=None, query=None):
+    def from_name(cls, reddit, name, loader, order=None, query=None,
+                  listing='r', period=None):
+
+        # Strip leading, trailing, and redundant backslashes
+        n = ''
+        n = ''.join([n + ''.join(list(g)) if k != '/' else '/' \
+                                       for k, g in groupby(name)]).strip(' /')
```

Comment:
`groupby()` is used here to group redundant `/`'s and flatten them. This is to catch those edge cases in the unit test. Also, typos like `/r/python//` or `/r/python//top` do what you'd want them to do.

Yes, this segment can be written more clearly.